woocommerce/woocommerce-ios
# Validate coupon percent amount

Summary
==========
Fix issue #6837 by taking advantage of the `Coupon Amount` input validation to enforce a valid percentage input. After going back and forth on how to approach this improvement. I decided to go with a slightly different solution from the one proposed in the issue discussion. Considering that it's a logical expectation that a coupon discount percentage can't be above 100%, the input validation might be more straightforward than allowing anything and handling an error notice later.

Screenshots
==========
https://user-images.githubusercontent.com/5920403/172292598-5bd49aaa-5dc2-42f3-bcf8-78f5467fb8f3.mp4



How to Test
==========
1. Make sure the Coupons experimental toggle is activated
2. Go to the Coupon list, select any editable Coupon
3. Inside the Coupon details, select the Edit coupon in the action menu
4. Once inside the edit view, change the Coupon discount type to percent
5. Verify that it's not possible to type any percent above 100 in there

Update release notes:
